app/ts/components/Transfer.tsx

```diff
@@ -129,14 +119,14 @@ export const Transfer = ({ provider, blockInfo }: { provider: Signal<ProviderSto
 		}
 		fetchingStates.value = 'fetching'
 		try {
-			const identifiedAddress = await itentifyAddress(address, id, provider.value?.provider, provider.value?.walletAddress)
-			if (identifiedAddress.address === contractAddress.value && identifiedAddress.inputId === itemId.value) {
-				if (identifiedAddress.type === 'ERC721') {
-					selectedNft.value = identifiedAddress
-				} else if (identifiedAddress.type === 'ERC1155') {
-					selectedNft.value = identifiedAddress
-				} else if (identifiedAddress.type === 'ERC20') {
-					warning.value = 'Token address provided is an ERC20 contract'
+			const identifiedAssets = await itentifyAddress(address, id, provider.value?.provider, provider.value?.walletAddress)
+			if (identifiedAssets[0].address === contractAddress.value && identifiedAssets[0].inputId === itemId.value) {
+				const supportedTypes = identifiedAssets.filter((token) => ['ERC721', 'ERC1155'].includes(token.type)) as SupportedToken[]
```

Comment:
```suggestion
				const supportedTypes = identifiedAssets.filter((token): token is SupportedToken=> ['ERC721', 'ERC1155'].includes(token.type))
```
could this be done without casting is ´is`?